torch/distributed/_tensor/placement_types.py

```diff
@@ -626,6 +626,29 @@ def dim_map(self) -> List[int]:
                 r[shard_dim] = i
         return r
 
+    @property
+    def num_shards_map(self) -> List[int]:
+        """
+        dim_map is a property we derive from `placements` of
```

Comment:
nit: I think we may prefer to start the comment with "what is `num_shards_map`?" directly (at least at a high level) and then compare it with `dim_map`, or do you think that it is a requirement for users to know what is `dim_map` first before understanding what is `num_shards_map`?